tests/exercise_instructions.rs

```diff
@@ -0,0 +1,501 @@
+                "Instruction meter did not consume expected amount"
+            );
+        }
+    };
+}
+
+macro_rules! test_interpreter_and_jit_asm {
+    ($source:expr, $config:expr, $mem:tt, ($($location:expr => $syscall_function:expr),* $(,)?), $context_object:expr $(,)?) => {
+        #[allow(unused_mut)]
+        {
+            let mut config = $config;
+            config.enable_instruction_tracing = true;
+            let mut function_registry = FunctionRegistry::<BuiltinFunction<TestContextObject>>::default();
+            $(test_interpreter_and_jit!(register, function_registry, $location => $syscall_function);)*
+            let loader = Arc::new(BuiltinProgram::new_loader(config, function_registry));
+            let mut executable = assemble($source, loader).unwrap();
+            test_interpreter_and_jit!(executable, $mem, $context_object);
+        }
+    };
+    ($source:expr, $mem:tt, ($($location:expr => $syscall_function:expr),* $(,)?), $context_object:expr $(,)?) => {
+        #[allow(unused_mut)]
+        {
+            test_interpreter_and_jit_asm!($source, Config::default(), $mem, ($($location => $syscall_function),*), $context_object);
+        }
+    };
+}
+
+// BPF_ALU : Arithmetic and Logic
+#[test]
+fn fuzz_alu() {
+    let seed = 0xC2DB2F8F282284A0;
+    let mut prng = SmallRng::seed_from_u64(seed);
+
+    for src in 0..10 {
+        for dst in 0..10 {
+            for _ in 0..10 {
+                test_ins(format!("mov64 r{src}, r{dst}"), &mut prng, 21);
+                test_ins(format!("add64 r{src}, r{dst}"), &mut prng, 21);
+                test_ins(format!("sub64 r{src}, r{dst}"), &mut prng, 21);
+                test_ins(format!("xor64 r{src}, r{dst}"), &mut prng, 21);
+                test_ins(format!("and64 r{src}, r{dst}"), &mut prng, 21);
+                test_ins(format!("lmul64 r{src}, r{dst}"), &mut prng, 21);
+                test_ins(format!("uhmul64 r{src}, r{dst}"), &mut prng, 21);
+                test_ins(format!("shmul64 r{src}, r{dst}"), &mut prng, 21);
+                test_ins(format!("udiv64 r{src}, r{dst}"), &mut prng, 21);
+                test_ins(format!("urem64 r{src}, r{dst}"), &mut prng, 21);
+                test_ins(format!("sdiv64 r{src}, r{dst}"), &mut prng, 21);
+
+                test_ins(format!("lsh64 r{src}, r{dst}"), &mut prng, 21);
+                test_ins(format!("rsh64 r{src}, r{dst}"), &mut prng, 21);
+                test_ins(format!("arsh64 r{src}, r{dst}"), &mut prng, 21);
+
+                test_ins(format!("mov32 r{src}, r{dst}"), &mut prng, 21);
+                test_ins(format!("add32 r{src}, r{dst}"), &mut prng, 21);
+                test_ins(format!("sub32 r{src}, r{dst}"), &mut prng, 21);
+                test_ins(format!("xor32 r{src}, r{dst}"), &mut prng, 21);
+                test_ins(format!("and32 r{src}, r{dst}"), &mut prng, 21);
+                test_ins(format!("lmul32 r{src}, r{dst}"), &mut prng, 21);
```

Comment:
SBFv1 instructions are missing, like `mod64`, `mul64`, `or64`, `div64` (and their 32-bit variants).
For SBFv2, I coudn't find `hor64`.

Reply:
| SBFv1 instructions are missing, like mod64, mul64, or64, div64

`mul64` => `lmul64`, `mod64` => `srem64`, `urem64`, `div64` => `sdiv64`, `udiv64`. 
I did miss `or64`, that's fixed now. 

|  (and their 32-bit variants).

I don't know what's missing.

| For SBFv2, I coudn't find hor64.

Line 460

Reply:
> | SBFv1 instructions are missing, like mod64, mul64, or64, div64
> 
> `mul64` => `lmul64`, `mod64` => `srem64`, `urem64`, `div64` => `sdiv64`, `udiv64`. I did miss `or64`, that's fixed now.

Even though the instructions do the same thing, they take a different code path for execution in the interpreter.

Reply:
added